### PR TITLE
Check signature in RemoveTask - nested messages

### LINKED ISSUE #2018: Anyone can cancel any task
The `RemoveTask` message contains only `task_id`. It's possible, that a rouge node could send a `RemoveTask` message for every task header it receives.

It can be mitigated by signing the `task_id` by the requestor's key. Or signing a string `"remove {}".format(task_id)`, to make sure it can't be extracted from some other message.

### PR DESCRIPTION
**Pull request by badb**
Alternative implementation of: https://github.com/golemfactory/golem/pull/2134
Please choose the version that you prefer. Personally, I vote for this one, but I'm not 100% sure, that it's the right approach. 

Resolves #2018

Should be merged with updated message version: https://github.com/golemfactory/golem-messages/pull/160

### CODE REVIEW

**Review by jiivan — APPROVED** [review]
LGTM (after messages release)

File: tests/golem/network/p2p/test_peersession.py [hunk 1]
```diff
@@ -489,6 +496,79 @@ def test_react_to_peers(self, add_peer_mock):
...
+    def test_react_to_remove_task_broadcast(self):
+        msg, task_id, previous_ka = \
+            self._gen_data_for_test_react_to_remove_task()
+        th_mock = mock.MagicMock()
+        keys_auth = self.peer_session.p2p_service.keys_auth
+        th_mock.task_owner_key_id = keys_auth.key_id
+        task_server = self.peer_session.p2p_service.task_server
+        task_server.task_keeper.task_headers[task_id] = th_mock
+        msg.serialize()
+        print(msg.sig)
```
Comment (etam) [inline tests/golem/network/p2p/test_peersession.py]: print?

**Review by etam — APPROVED** [review]
LGTM, just one small `print` was left behind

**Comment by badb** [comment]
Blocked by: https://github.com/golemfactory/golem/pull/2291

**Pull request merged**